tests/test_lib_smsprovider.py

```diff
@@ -256,6 +267,35 @@ def test_08_smsgateway_success(self):
         r = sms.submit_message("123456", "Halo")
         self.assertTrue(r)
 
+    @smtpmock.activate
+    def test_09_send_sms_regexp_success(self):
+        smtpmock.setdata(response={"recp@example.com": (200, "OK")})
+        # Here we need to send the SMS
+        r = self.regexp_provider.submit_message("+49 123/456-78", "Hello")
+        self.assertTrue(r)
+
+        p = SmtpSMSProvider._mangle_phone("+49 123/456-78", self.regexp_config)
+        self.assertEqual("4912345678", p)
```

Comment:
These tests just check the static code, they can be moved to the `SMSTestCase` class.

Reply:
https://github.com/privacyidea/privacyidea/pull/3597/commits/55a986bd6306f3f9b353e3a54bb87360d5601561